account_invoice_pricelist/tests/test_account_move_pricelist.py

```diff
@@ -316,3 +334,9 @@ def test_check_currency(self):
             self.invoice.with_context(force_check_currecy=True).write(
                 {"currency_id": self.usd_currency.id}
             )
+
+    def test_account_invoice_creditnote_line_qty_flip_when_total_negative(self):
+        self.negative_invoice.action_switch_invoice_into_refund_credit_note()
+        invoice_line = self.negative_invoice.invoice_line_ids[:1]
+        self.assertAlmostEqual(invoice_line.price_unit, 100.00)
+        self.assertEqual(invoice_line.discount, 0.00)
```

Comment:
I suppose we don't necessarily have to have the negative invoice in the setup.

```suggestion
        self.invoice.pricelist_id = self.sale_pricelist
        self.invoice.invoice_line_ids.write({"quantity": -1, "price_unit": 100.00})
        self.invoice.action_switch_invoice_into_refund_credit_note()
        for invoice_line in self.invoice.invoice_line_ids
            self.assertAlmostEqual(invoice_line.price_unit, 100.00)
```

If we keep the negative invoice, we should still assign a pricelist to it.